org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/EventHub.java

```diff
@@ -60,15 +60,27 @@ public void start(VirtualMachine vm) {
                     }
 
                     EventSet set = queue.remove();
+
+                    // Print JDI Events
+                    StringBuffer buf = new StringBuffer("\nJDI Event Set: {");
+                    for (Event event : set) {
+                        buf.append(event);
+                        buf.append(", ");
+                    }
+                    buf.append("}\n");
+                    Logger.logInfo(buf.toString());
+
                     boolean shouldResume = true;
+                    boolean beConsumed = false;
```

Comment:
VM will trigger two kinds of event, one is system event such as VMStartEvent, VMDeathEvent, VMDisconnectEvent, and another is event requested by user such as BreakpointEvent, StepEvent, ThreadStartEvent, ThreadDeathEvent etc. For system event, it's OK to invoke eventSet.resume(). For user registered event, of course they should provide handler to resolve the event.

So you're right, it's OK to resume by default. Will remove consumed flag.